h2/src/main/org/h2/util/ValueHashMap.java

```diff
@@ -174,6 +191,42 @@ public V get(Value key) {
         }
         return list;
     }
+    
+    public Iterable<Map.Entry<Value, V>> entries() {
+        return new EntryIterable();
+    }
+    
+    private final class EntryIterable implements Iterable<Map.Entry<Value, V>> {
```

Comment:
An explicit no-arg constructor with default visibility in needed here to avoid synthetic access. It will replace automatic private constructor without any additional overhead.